lib/src/main/kotlin/com/swmansion/starknet/provider/Provider.kt

```diff
@@ -841,7 +841,7 @@ interface Provider {
      * @param simulationFlags set of flags to be used for simulation
      * @return a list of transaction simulations
      */
-    fun simulateTransactions(transactions: List<TransactionPayload>, blockTag: BlockTag, simulationFlags: Set<SimulationFlag>): Request<SimulatedTransactionList>
+    fun simulateTransactions(transactions: List<Transaction>, blockTag: BlockTag, simulationFlags: Set<SimulationFlag>): Request<SimulatedTransactionList>
```

Comment:
I don't think we're supposed to be able to simulate / estimate fees / etc. for **every** type of transaction. As for example, there are outdated `V0`, `V1` transactions, that we should able to fetch from chain, but not execute. 

I think we should have an interface that determines if transaction is **executable** (can be treated as a payload and sent for execution), similarly to how it's done in swift, so transactions that are not meant to be executed are not passed in these methods.

Reply:
I added `ExecutableTransaction` interface and used in in `simulateTransaction` and `getEstimateFee` in provider.